src/tests/electron/tests/android-setup-prompt-grant-permissions.test.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import {
+    leftFooterButtonAutomationId,
+    rightFooterButtonAutomationId,
+} from 'electron/views/device-connect-view/components/android-setup/android-setup-step-layout';
+import { tryAgainAutomationId } from 'electron/views/device-connect-view/components/android-setup/prompt-grant-permissions-step';
+import { getAutomationIdSelector } from 'tests/common/get-automation-id-selector';
+import { createApplication } from 'tests/electron/common/create-application';
+import { scanForAccessibilityIssues } from 'tests/electron/common/scan-for-accessibility-issues';
+import { AndroidSetupViewController } from 'tests/electron/common/view-controllers/android-setup-view-controller';
+import { AppController } from 'tests/electron/common/view-controllers/app-controller';
+import {
+    commonAdbConfigs,
+    delayAllCommands,
+    setupMockAdb,
+    simulateServiceLacksPermissions,
+} from '../../miscellaneous/mock-adb/setup-mock-adb';
+
+describe('Android setup - prompt-grant-permissions', () => {
+    const [cancelId, nextId] = [leftFooterButtonAutomationId, rightFooterButtonAutomationId];
+    const defaultDeviceConfig = commonAdbConfigs['single-device'];
+    let app: AppController;
+    let dialog: AndroidSetupViewController;
+
+    beforeEach(async () => {
+        await setupMockAdb(simulateServiceLacksPermissions(defaultDeviceConfig));
+        app = await createApplication({ suppressFirstTimeDialog: true });
+        dialog = await app.openAndroidSetupView('prompt-grant-permissions');
+    });
+
+    afterEach(async () => {
+        if (app != null) {
+            await app.stop();
+        }
+    });
+
+    it('initial component state is correct', async () => {
+        expect(await dialog.isEnabled(getAutomationIdSelector(cancelId))).toBe(true);
+        expect(await dialog.isEnabled(getAutomationIdSelector(nextId))).toBe(false);
+        expect(await dialog.isEnabled(getAutomationIdSelector(tryAgainAutomationId))).toBe(true);
+    });
+
+    it('goes to prompt-choose-device upon cancel', async () => {
+        await setupMockAdb(commonAdbConfigs['multiple-devices']);
+        await dialog.client.click(getAutomationIdSelector(cancelId));
+        await dialog.waitForDialogVisible('prompt-choose-device');
+    });
+
+    it('try again returns here if permissions are not granted', async () => {
+        await setupMockAdb(
+            delayAllCommands(5000, simulateServiceLacksPermissions(defaultDeviceConfig)),
+        );
+        await dialog.client.click(getAutomationIdSelector(tryAgainAutomationId));
+        await dialog.waitForDialogVisible('detect-permissions');
+        await dialog.waitForDialogVisible('prompt-grant-permissions');
+    });
+
+    it('try again moves on if permissions are granted; detect-permissions a11y test', async () => {
+        await setupMockAdb(delayAllCommands(2500, commonAdbConfigs['single-device']));
```

Comment:
Can we leverage defaultDeviceConfig, so there's one place to change it we change it?
```suggestion
        await setupMockAdb(delayAllCommands(2500, defaultDeviceConfig));
```

Reply:
Will fix in upcoming PR